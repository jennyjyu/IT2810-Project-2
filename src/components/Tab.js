// Src for inspiration for tabs: https://alligator.io/react/tabs-component/
import React, {Component} from 'react';
import PropTypes from 'prop-types';


class Tab extends Component {
  static propTypes = {
    activeTab: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = {
    }
    this.onClick = this.onClick.bind(this);
  }

  onClick() {
    const { label, onClick } = this.props;
    onClick(label);
  }

  // onClick = () => {
  //   const { label, onClick } = this.props;
  //   onClick(label);
  // }

  render() {
    const {
      onClick,
      props: {
        activeTab,
        label,
      },
    } = this;

    let className = 'tab-list-item';

    if (activeTab === label) {
      className += ' tab-list-active';
    }

    return (
      <li
        className={className}
        onClick={onClick}
      >
        {label}
      </li>
    );
  }
}

export default Tab;
