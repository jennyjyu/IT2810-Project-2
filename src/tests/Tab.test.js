import React from "react"
import Tab from "../components/Tab"
import renderer from "react-test-renderer"
// import App from "../App"



test("Tab snapshot test", () => {
  const component = renderer.create(
    <Tab/>
  )
  let tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})
