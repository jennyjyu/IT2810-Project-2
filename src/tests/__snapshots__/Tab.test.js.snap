// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab snapshot test 1`] = `
<li
  className="tab-list-item tab-list-active"
>
  Tab 
</li>
`;
